### ISSUE TITLE
All the tests will re-run even if a single test failed

### ISSUE THREAD

**Issue by rakeshnambiar** [yahoo/protractor-retry#87]
All the tests will re-run even if a single test failed. I am using `serenity-js framework` [chai assertions].

Configuration:
```javascript
    onCleanUp: function (results) {
        retry.onCleanUp(results);
    },
    onPrepare: function () {
        retry.onPrepare();
    },
    afterLaunch: function() {
        return retry.afterLaunch(1);
    }
```

**Comment by rakeshnambiar** [comment]
@jan-molak any thoughts?

**Comment by rakeshnambiar** [comment]
@dreuxl @amrot17 Please take a look at this issue.

**Comment by jan-molak** [comment]
@rakeshnambiar if you're using Serenity/JS with the [Mocha adapter](https://serenity-js.org/modules/mocha), you can tell Mocha to [retry the failed tests](https://serenity-js.org/modules/mocha/class/src/adapter/MochaConfig.ts~MochaConfig.html#instance-member-retries), therefore removing the need for `protractor-retry`.

I'll be adding support for retrying failed Cucumber tests soon.

**Comment by dreuxl** [comment]
@rakeshnambiar are u running your suite in parallel , let s say using sauce labs? 
https://github.com/yahoo/protractor-retry#known-caveat
If not it s expected.

Also note, the mocha retry may address your issue, even though the mechanism is different:
* mocha is retrying on the fly (same browser/session)
* protractor-retry is retrying at the end of a suite and create a new session (clean environment) for the retry

**Comment by rakeshnambiar** [comment]
@dreuxl no, the execution in the sequential order on selenium-grid. I am not using the Mocha runner and it's a cucumber test by using `serenity-js framework`. Probably like @jan-molak mentioned, I will have to wait.

**Comment by idem7x** [comment]
I think this could happen if you have your specs inside capabilities. Moving it outside capabilities will solve issue.
